lite/errors/errors.go

```diff
@@ -3,90 +3,110 @@ package errors
 import (
 	"fmt"
 
-	"github.com/pkg/errors"
+	cmn "github.com/tendermint/tendermint/libs/common"
 )
 
-var (
-	errValidatorsChanged = fmt.Errorf("Validators differ between header and certifier")
-	errCommitNotFound    = fmt.Errorf("Commit not found by provider")
-	errTooMuchChange     = fmt.Errorf("Validators change too much to safely update")
-	errPastTime          = fmt.Errorf("Update older than certifier height")
-	errNoPathFound       = fmt.Errorf("Cannot find a path of validators")
-)
+//----------------------------------------
+// Error types
+
+type errCommitNotFound struct{}
 
-// IsCommitNotFoundErr checks whether an error is due to missing data
-func IsCommitNotFoundErr(err error) bool {
-	return err != nil && (errors.Cause(err) == errCommitNotFound)
+func (e errCommitNotFound) Error() string {
+	return "Commit not found by provider"
 }
 
-// ErrCommitNotFound indicates that a the requested commit was not found.
-func ErrCommitNotFound() error {
-	return errors.WithStack(errCommitNotFound)
+type errUnexpectedValidators struct {
```

Comment:
Not sure if it helps returning hashes of different valset.

Reply:
I don't know the answer, but I know I want to punt these kinds of tweaks for now.